packages/flutter_tools/lib/src/commands/migrate_resolve_conflicts.dart

```diff
@@ -0,0 +1,322 @@
+
+    for (int i = 0; i < conflictFiles.length; i++) {
+      final String localPath = conflictFiles[i];
+      final File file = stagingDirectory.childFile(localPath);
+      final List<String> lines = file.readAsStringSync().split('\n');
+      // We write a newline in the output, this counteracts it.
+      if (lines.last == '') {
+        lines.removeLast();
+      }
+
+      // Find all conflicts
+      final List<Conflict> conflicts = findConflicts(lines);
+
+      // Prompt developer
+      await promptDeveloperSelectAction(conflicts, lines, localPath);
+
+      final bool result = await verifyAndCommit(conflicts, lines, file, localPath);
+      if (!result) {
+        i--;
+      }
+    }
+    return const FlutterCommandResult(ExitStatus.success);
+  }
+
+  /// Parses the lines of a file and extracts a list of Conflicts.
+  List<Conflict> findConflicts(List<String> lines) {
+    // Find all conflicts
+    final List<Conflict> conflicts = <Conflict>[];
+    Conflict currentConflict = Conflict.empty();
+    for (int lineNumber = 0; lineNumber < lines.length; lineNumber++) {
+      final String line = lines[lineNumber];
+      if (line.contains(_conflictStartMarker)) {
+        currentConflict.startLine = lineNumber;
+      } else if (line.contains(_conflictDividerMarker)) {
+        currentConflict.dividerLine = lineNumber;
+      } else if (line.contains(_conflictEndMarker)) {
+        currentConflict.endLine = lineNumber;
+        assert(
+          (currentConflict.startLine == null || currentConflict.dividerLine == null || currentConflict.startLine! < currentConflict.dividerLine!) &&
+          (currentConflict.dividerLine == null || currentConflict.endLine == null || currentConflict.dividerLine! < currentConflict.endLine!)
+        );
+        conflicts.add(currentConflict);
+        currentConflict = Conflict.empty();
+      }
+    }
+    return conflicts;
+  }
+
+  /// Display a detected conflict and prompt the developer on whether to accept the original lines, new lines,
+  /// or skip handling the conflict.
+  Future<void> promptDeveloperSelectAction(List<Conflict> conflicts, List<String> lines, String localPath) async {
+    final int contextLineCount = int.parse(stringArg('context-lines')!);
+    for (final Conflict conflict in conflicts) {
+      if (!conflict.isValid) {
+        conflict.keepOriginal = null;
+        continue;
+      }
+      // Print the conflict for reference
+      logger.printStatus(terminal.clearScreen(), newline: false);
+      logger.printStatus('Cyan', color: TerminalColor.cyan, newline: false);
+      logger.printStatus(' = Original lines.  ', newline: false);
+      logger.printStatus('Green', color: TerminalColor.green, newline: false);
+      logger.printStatus(' = New lines.\n', newline: true);
+
+      // Print the conflict for reference
+      for (int lineNumber = (conflict.startLine! - contextLineCount).abs(); lineNumber < conflict.startLine!; lineNumber++) {
```

Comment:
Shouldn't this be, instead of getting absolute value of the difference, getting the greater of the difference and 0?